### ISSUE TITLE
Browser: crashes on visiting http://bettermotherf*ckingwebsite.com

### ISSUE THREAD

**Issue by boricj** [SerenityOS/serenity#687]
Since Browser can successfully render http://motherfuckingwebsite.com, the next logical step would be to successfully render http://bettermotherfuckingwebsite.com.

There are two issues when trying to visit this website:
- The webserver always sends payloads with `Content-Encoding: gzip` and seems to disregard any kind of `Accept-Encoding` header. The UTF-8 parser is not happy with the binary data, handling gzip-compressed replies is required for any practical web surfing experience these days.
- Using an uncompressed local copy, the CSS parser dies with:

```
Browser(20): CLocalSocket{0x01174016} connecting to /tmp/wsportal...
Browser(20): CLocalSocket{0x01174016} connected ok!
WindowServer(9): [WM] WSWindow{0x017ab016} title set to "Browser"
Browser(20): HtmlView::load: file:///home/anon/index.html
Browser(20): Unhandled escape sequence
Browser(20): Unhandled escape sequence
Browser(20): CSS PARSER ASSERTION FAILED: peek() == ch
Browser(20): At character# 120 in CSS: _body{margin:40px
auto;max-width:650px;line-height:1.6;font-size:18px;color:#444;padding:0
10px}h1,h2,h3{line-height:1.2}_
USERSPACE(20) ASSERTION FAILED: false
Parser/CSSParser.cpp:126 in char CSSParser::consume_specific(char)
signal: Browser(20) sent 6 to Browser(20)
```

As it is a fairly simple website and the W3C validator finds no errors on that webpage, it seems like a reasonable test case to improve the web browser on.

**Comment by schnema123** [comment]
The crash is because '}' is a valid character for property values in [1] and is therefore consumed by parse_property(). This is because I think the CSS parsers expects that a property declaration is ending with a ';', which is not the case in the given CSS [2].

[1] https://github.com/SerenityOS/serenity/blob/eb77e680ed615e9daf5b165359b4a45541b4e6f3/Libraries/LibHTML/Parser/CSSParser.cpp#L283
[2]
```
        h1,
        h2,
        h3 {
            line-height: 1.2
        }
```

**Comment by linusg** [comment]
Andreas did a video on this:

> Browser hacking: Fixing bugs on "better motherfricken website"
> https://www.youtube.com/watch?v=KsiclBmE4Pg

The gzip issue remains, but a local copy of http://bettermotherfuckingwebsite.com does render now :smile:

**Comment by schnema123** [comment]
I am working on gzip at the moment. Trying to understand the project structure to know where to put stuff...

**Comment by awesomekling** [comment]
Fixed as of c40935e79f6a404e053b1c5908d8acf557cfaa91, thanks to @schnema123 :tada:
